src/components/typeahead/typeahead.spec.ts

```diff
@@ -0,0 +1,143 @@
+
+@Component({
+    selector: 'app-typeahead-test',
+    template: `
+        <div class="row" [class.drop-direction-up]="dropDirection === 'up'">
+            <div class="col-md-12 has-feedback">
+                <form class="input-typeahead-form">
+                    <input [(ngModel)]="input"
+                           [ngModelOptions]="{standalone: true}"
+                           placeholder="Enter Text"
+                           class="form-control"
+                           type="text"
+                           (click)="dropdownOpen = true"
+                           (keydown)="typeaheadKeyService.handleKey($event, typeahead)"
+                           (keydown.escape)="dropdownOpen = false"
+                           (blur)="dropdownOpen = false">
+
+                    <ux-typeahead #typeahead
+                                  class="typeahead-example"
+                                  [(open)]="dropdownOpen"
+                                  [filter]="input"
+                                  [options]="loadOptionsFn"
+                                  [openOnFilterChange]="true"
+                                  [pageSize]="pageSize"
+                                  [selectOnEnter]="selectOnEnter"
+                                  [selectFirst]="selectFirst"
+                                  [dropDirection]="dropDirection"
+                                  (optionSelected)="input = $event.option"
+                                  [(recentOptions)]="recentOptions"
+                                  [recentOptionsMaxCount]="recentOptionsMaxCount">
+                    </ux-typeahead>
+                </form>
+            </div>
+        </div>
+    `
+})
+export class TypeaheadTestComponent {
+
+    values: ReadonlyArray<string> = [];
+
+    dropdownOpen: boolean = false;
+    selectOnEnter: boolean = true;
+    dropDirection: 'down' | 'up' | 'auto' = 'auto';
+    selectFirst: boolean = true;
+    recentOptions: ReadonlyArray<string>;
+    recentOptionsMaxCount: number = 5;
+    maxHeight: string;
+    pageSize: number = 40;
+
+    input: string = '';
+
+    loadOptionsFn = this.loadOptions.bind(this);
+
+    /** Load the options and filter the them */
+    loadOptions(pageNum: number, pageSize: number, filter: string): Promise<ReadonlyArray<string>> {
+
+        // get the values for the current page based on the filter text provided
+        const values = this.values.filter(tag => tag.toLowerCase().indexOf(filter.toLowerCase()) !== -1)
+            .slice(pageNum * pageSize, (pageNum + 1) * pageSize);
+
+        // return the values after a delay to simulate server response time
+        return of(values).pipe(delay(1000)).toPromise();
+    }
+
+    constructor(public typeaheadKeyService: TypeaheadKeyService<string>) {
+
+        /* Adding values to typeahead list */
+        for (let index = 0; index < 200; index++) {
+            this.values = [...this.values, chance.name()];
```

Comment:
I know this doesn't affect the tests as they currently stand, but we should never use chance in a testing scenario. If you could replace it with something deterministic here that'd be great.